matrix/sherman_morrison.py

```diff
@@ -30,39 +37,73 @@ def __str__(self):
         string_format_identifier = "%%%ds" % (max_element_length,)
 
         # Make string and return
-        for row_vector in self.array:
-            s += "["
-            s += ", ".join(string_format_identifier % (obj,) for obj in row_vector)
-            s += "]\n"
+        s += "\n".join("[" + ", ".join(string_format_identifier % (obj,) for obj in row_vector) + "]" for row_vector in self.array)
```

Comment:
This line can not be read on GitHub without left-right scrolling.  Please wrap it so this is no longer the case.